cmd/snappy-app-dev/main.go

```diff
@@ -0,0 +1,197 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"log/syslog"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/snap"
+)
+
+var deviceCgroupDir = `/sys/fs/cgroup/devices`
+
+func verifyAction(action string) (string, error) {
+	switch action {
+	case
+		"add",
+		"change",
+		"remove":
+		return action, nil
+	}
+	return "", fmt.Errorf("unsupported action %q", action)
+}
+
+func verifyAppname(appname string) (string, error) {
```

Comment:
I think that `appname` should be something different not to confuse with an actual application name. We use the term snap security tag. Perhaps we just want to use "snap udev security tag" as a specialization of that term?